File: games_services/android/src/main/kotlin/com/abedalkareem/games_services/models/AchievementItemData.kt
package com.abedalkareem.games_services.models

data class AchievementItemData(
  val id: String,
  val name: String,
  val description: String,
  val lockedImage: String?,
  val unlockedImage: String?,
  val completedSteps: Int,
  val totalSteps: Int,
  val unlocked: Boolean,
)

File: games_services/darwin/games_services/Sources/games_services/Models/AchievementItemData.swift
import Foundation

struct AchievementItemData: Codable {
  var id: String
  var name: String
  var description: String
  var lockedImage: String?
  var unlockedImage: String?
  var completedSteps: Int
  var totalSteps: Int = 100
  var unlocked: Bool
}
